src/server/api/lib/middleware/authorization.middleware.js

```diff
@@ -3,18 +3,22 @@ const { getAuthToken } = require('./getToken.backend');
 const knex = require('../../../config/db');
 const Error = require('../utils/http-error');
 
+const selectUserAndRole = async (userId) => {
+  const result = await knex('users')
+    .join('user_roles', 'user_roles.fk_user_id', '=', 'users.id')
+    .join('roles', 'roles.id', '=', 'user_roles.fk_role_id')
+    .select('users.uid', 'roles.name')
+    .where('users.uid', userId);
+  return result[0];
+};
+
```

Comment:
If you are looking for only one record from the database consider using `.first()` instead of accessing the array with an index.

Read here: http://knexjs.org/#Builder-first

Reply:
changed :)